### ISSUE TITLE
[apex] Results interleave with logging on CLI to create invalid XML output

### ISSUE THREAD

**Issue by jfeingold35** [pmd/pmd#5131]
**Affects PMD Version:** 7.3.0

**Description:**

Scanning certain files causes console log information to incorrectly interleave with the output as it's being written to the CLI, which makes the output invalid and unparseable.

**Exception Stacktrace:**

N/A. Does not cause an exception.

**Code Sample demonstrating the issue:**

```apex
public with sharing class PseudoNamespace_ClassName {
    @InvocableMethod(Label='Write To Integration Provider' Category='Notes' Callout=true)
    public static List<String> writeNote(List<Parameters> paramsList) {
        List<String> returnStrings = new List<String>();
        List<VCC_Progress_Note__c> pnList = new List<VCC_Progress_Note__c>();
        for (Parameters params : paramsList) {
            switch on params.theObjectWeManipulateInTheInvocable {
                when TypeOne castVarOne {
                    // The contents of the when block don't actaully matter for this bug.
                }
                when TypeTwo CastVarTwo {
                    // The contents of the when block don't actaully matter for this bug.
                }
            }
        }
    }

    /**
    * @description Inner class representing the parameters being passed into to the parent invocable.
    */
    public class Parameters {
        @InvocableVariable
        public CustomMetadata__mdt configurationCustomMetadata;

        @InvocableVariable
        public SObject theObjectWeManipulateInTheInvocable;

        @InvocableVariable
        public PseudoNamespace_ConfigObjectOne paramsOne;

        @InvocableVariable
        public PseudoNamespace_ConfigObjectTwo paramsTwo;
    }
}

```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1. Download and unzip the latest version of PMD.
2. Save the above Apex code as `PseudoNamespace_ClassName.cls`
3. Run `bin/pmd check -d path/to/PseudoNamespace_ClassName.cls --format xml --rulesets category/apex/bestpractices.xml`
4. Observe that the opening `<pmd>` tag is:
```xml
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd" version="7.2.0" timestamp="2024-07-23T11:38:[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_userdocs_incremental_analysis.html
50.876"No adapter exists for type com.google.summit.ast.declaration.VariableDeclarationGroup
No adapter exists for type com.google.summit.ast.declaration.VariableDeclarationGroup
>
```
The "No adapter..." is a floating bit of log that is somehow getting injected into the output, thus rendering it invalid XML.
Note that if you add `-r o.xml`, then the log writes to the console and the output writes to the outfile and is syntactically valid.

EXPECTED BEHAVIOR: The log should appear above the output.

**Running PMD through:** *CLI*

**Comment by adangel** [comment]
Thanks for reporting.
This is a Duplicate of #5094 and will be fixed with PMD 7.4.0.

**Comment by adangel** [comment]
Also note: Use the `--report-file` cli parameter instead of redirecting stdout...
